src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -656,8 +668,16 @@ public function getBitwiseXorType(Expr $left, Expr $right, callable $getTypeCall
 	 */
 	public function getSpaceshipType(Expr $left, Expr $right, callable $getTypeCallback): Type
 	{
-		$leftTypes = TypeUtils::getConstantScalars($getTypeCallback($left));
-		$rightTypes = TypeUtils::getConstantScalars($getTypeCallback($right));
+		$callbackLeftType = $getTypeCallback($left);
+		$callbackRightType = $getTypeCallback($right);
+
+		if ($callbackLeftType instanceof NeverType || $callbackRightType instanceof NeverType) {
+			return new NeverType();
+		}
+
+		$leftTypes = TypeUtils::getConstantScalars($callbackLeftType);
+		$rightTypes = TypeUtils::getConstantScalars($callbackRightType);
```

Comment:
what do you think of also calling them `$leftType` and `$rightType`? I think then it would be consistent with other methods from here

Reply:
it collides with a already same named variable in the loop below